docs/src/philosophy.md

```diff
@@ -0,0 +1,127 @@
+
+However, being a `Real` means way more than just being compatible with
+arithmetic operations.
+`Real`s are also expected to
+
+1. Be compatible with any other `Number` through promotion.
+2. Support comparison operations, such as `==` or `<`.
+3. Act as a container of a single element,
+   e.g. `collect(x)` returns a 0-dimensional array containing `x`.
+
+Each of those points lead to specific design choice for `IntervalArithmetic.jl`,
+choices that we detail below.
+
+
+## Compatibility with other `Number`s
+
+In julia it is expected that `1 + 2.2` silently promoted the integer `1`
+to a `Float64` to be able to perform the addition.
+Following this logic, it means that `0.1 + interval(2.2, 2.3)` should 
+silently promote `0.1` to an interval.
+
+However, in this case we can not guarantee that `0.1` is known exactly,
+because we do not know how it was produced in the first place.
+Following the julia convention is thus in contradiction with providing
+guaranteed result.
+
+In this case, we choose to be mostly silent,
+the information that a non-interval of unknown origin is recorded in the `NG` flag,
+but the calculation is not interrupted and no warning is printed.
+
+
+## Comparison operators
+
+We can extend our above definition of the desired behavior for two real numbers
+`x` and `y`, and their respective intervals `X` and `Y`.
+With this, we want to have, for any function`f`,
+for all `x` in `X` and all `y` in `Y`,
+```julia
+f(x, y) ∈ f(X, Y)
+```
+
+With this in mind, an operation such as `==` can easily be defined for intervals
+
+1. If the intervals are disjoints (`X ∩ Y === ∅`), then `X == Y` is `[false]`.
+2. If the intervals both contain a single element,
+   and that element is the same for both,
+   `X == Y` is `[true]`.
+3. Otherwise, we can not conclude anything, and `X == Y` must be `[false, true]`.
+
+Not that we use intervals in all case, because, according to our definition,
+the true result must be contained in the returned interval.
+However, this is not convenient, as any `if` statement would error when used
+with an interval.
+Instead, we have opted to return respectively `false` and `true`
+for cases 1 and 2, and to immediately error otherwise.
+
+In this way, we can return a more informative error,
+but we only do it when the result is ambiguous.
+
+This has a clear cost, however, in that some expected behaviors do not hold.
+For example, an `Interval` is not equal to itself.
+
+```julia> X = interval(1, 2)
+[1.0, 2.0]_com
+
+julia> X == X
+ERROR: ArgumentError: `==` is purposely not supported when the intervals are overlapping. See instead `isequal_interval`
+Stacktrace:
+ [1] ==(x::Interval{Float64}, y::Interval{Float64})
+   @ IntervalArithmetic C:\Users\Kolaru\.julia\packages\IntervalArithmetic\XjBhk\src\intervals\real_interface.jl:86
+ [2] top-level scope
+   @ REPL[6]:1.
+```
+
+
+## Intervals as sets
+
+We have taken the perspective to always let `Interval`s act as if they were numbers.
```

Comment:
If we want to not avoid the word "number":

"
We have taken the perspective to always let `Interval`s behave as numerical values.
"